sdlib/d/gc/bin.d

```diff
@@ -153,3 +164,64 @@ private:
 		return current;
 	}
 }
+
+unittest batchAllocation {
```

Comment:
This is not testing interesting behavior, such as what happens when there are several partially filled slab in the bin.

Reply:
This is the problem that is being dealt with here.

Reply:
All the test are filling up available extents. We should see extents left behind and ensure that they are in the correct state.